### ISSUE TITLE
Pass row property to LegendFormatter callback

### ISSUE THREAD

**Issue by timeu** [danvk/dygraphs#726]
The new LegendFormatter callback is a nice way to customize the legends, however currently only the `x` value is passed to the callback inside the `data` object. It would be quite useful to also pass the `row` value (i.e. if one needs to do an indexed array acess into a `DataTable`  or `DataView` to retrieve additional information for a specific row). 
I can work on a pull-request if this is an acceptable feature request

**Comment by danvk** [comment]
Yes, I think that would be a fine addition. In the meantime, you can just call `Dygraph.prototype.getRowForX`.

**Comment by timeu** [comment]
Thanks for the workaround (although performance might be an issue). 
Great than I will go ahead and create the pull-request.

**Comment by danvk** [comment]
`getRowForX` does a binary search over the x-coordinates, so performance shouldn't be too bad. But of course passing the row directly would be even better.
